tests/file/test_handle.py

```diff
@@ -0,0 +1,153 @@
+# license information.
+# --------------------------------------------------------------------------
+import unittest
+
+from azure.storage.file import (
+    FileService,
+)
+from tests.testcase import (
+    StorageTestCase,
+    record,
+    TestMode,
+)
+
+# ------------------------------------------------------------------------------
+TEST_SHARE_NAME = 'test'
+
+
+# ------------------------------------------------------------------------------
+
+class StorageHandleTest(StorageTestCase):
+    def setUp(self):
+        super(StorageHandleTest, self).setUp()
+        self.fs = self._create_storage_service(FileService, self.settings)
+
+    def tearDown(self):
+        return super(StorageHandleTest, self).tearDown()
+
+    def _validate_handles(self, handles):
+        # Assert
+        self.assertIsNotNone(handles)
+        self.assertGreaterEqual(len(handles), 1)
+        self.assertIsNotNone(handles[0])
+
+        # verify basic fields
+        # path may or may not be present
+        # last_connect_time_string has been missing in the test
+        self.assertIsNotNone(handles[0].handle_id)
+        self.assertIsNotNone(handles[0].file_id)
+        self.assertIsNotNone(handles[0].parent_id)
+        self.assertIsNotNone(handles[0].session_id)
+        self.assertIsNotNone(handles[0].client_ip)
+        self.assertIsNotNone(handles[0].open_time)
+
+    @record
+    def test_list_handles_on_share(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True))
+
+        # Assert
+        self._validate_handles(handles)
+
+    @record
+    def test_list_handles_with_marker(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handle_generator = self.fs.list_handles(TEST_SHARE_NAME, recursive=True, max_results=1)
+
+        # Assert
+        self.assertIsNotNone(handle_generator.next_marker)
+        handles = list(handle_generator)
+        self._validate_handles(handles)
+
+        # Note down a handle that we saw
+        old_handle = handles[0]
+
+        # Continue listing
+        remaining_handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True, marker=handle_generator.next_marker))
+        self._validate_handles(handles)
+
+        # Make sure the old handle did not appear
+        # In other words, the marker worked
+        old_handle_not_present = all([old_handle.handle_id != handle.handle_id for handle in remaining_handles])
+        self.assertTrue(old_handle_not_present)
+
+    @record
+    def test_list_handles_on_directory(self):
```

Comment:
Missing tests for snapshot

Reply:
Sure. I wasn't sure how to hold handles for a snapshot, but just figured it out.